# 2023 DataMining Team3

## 우울팀

### 1. Import Library

In [26]:
import pandas as pd

### 2. Data preprocessing

#### 데이터 불러오기

In [27]:
csv_all = pd.read_csv("data/hn18_all.csv")
csv_description = pd.read_csv("description/hn18_all_description.csv")

/Users/subinpark/opt/anaconda3/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (146,209,220,274,283,318,339,355,436,555,570,572,587,647) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [28]:
csv_all

mod_d          ID    ID_fam  year  region  town_t  apt_t   psu  \
0     2023.01.13.  A701020501  A7010205  2018       1       1      2  A701   
1     2023.01.13.  A701021301  A7010213  2018       1       1      2  A701   
2     2023.01.13.  A701021302  A7010213  2018       1       1      2  A701   
3     2023.01.13.  A701022601  A7010226  2018       1       1      2  A701   
4     2023.01.13.  A701023303  A7010233  2018       1       1      2  A701   
...           ...         ...       ...   ...     ...     ...    ...   ...   
7987  2023.01.13.  R708028902  R7080289  2018       8       1      2  R708   
7988  2023.01.13.  R708029201  R7080292  2018       8       1      2  R708   
7989  2023.01.13.  R708029202  R7080292  2018       8       1      2  R708   
7990  2023.01.13.  R708029203  R7080292  2018       8       1      2  R708   
7991  2023.01.13.  R708029204  R7080292  2018       8       1      2  R708   

      sex  age  ...  HEI_TVEG   HEI_VEG  HEI_PROTF  HEI_DAIRY    HEI_SFA  \
0       1   80  ...  1.367163  0.775022   5.800000       0.00   5.273475   
1       1   78  ...  4.881043  1.641812  10.000000      10.00  10.000000   
2       2   75  ...  3.367648  4.943629   4.849725      10.00   0.000000   
3       2   43  ...       NaN       NaN        NaN        NaN        NaN   
4       2   80  ...  4.619405  0.437024   1.951565       0.00  10.000000   
...   ...  ...  ...       ...       ...        ...        ...        ...   
7987    2   73  ...  5.000000  3.460232   2.940008       7.28   6.332807   
7988    1   43  ...  5.000000  5.000000  10.000000       0.00   0.000000   
7989    2   39  ...  5.000000  5.000000  10.000000       0.00   0.000000   
7990    1   10  ...       NaN       NaN        NaN        NaN        NaN   
7991    2    8  ...       NaN       NaN        NaN        NaN        NaN   

         HEI_NA  HEI_SWEET   HEI_CHO   HEI_FAT    HEI_EN  
0     10.000000   10.00000  5.000000  5.000000  0.000000  
1      3.418798   10.00000  4.345249  2.956568  5.000000  
2      9.806158   10.00000  5.000000  5.000000  5.000000  
3           NaN        NaN       NaN       NaN       NaN  
4      8.029505   10.00000  0.000000  0.000000  1.244134  
...         ...        ...       ...       ...       ...  
7987  10.000000    7.98486  2.797653  5.000000  0.000000  
7988   2.017411   10.00000  0.000000  0.000000  0.000000  
7989   5.828002   10.00000  0.000000  0.000000  1.288707  
7990        NaN        NaN       NaN       NaN       NaN  
7991        NaN        NaN       NaN       NaN       NaN  

[7992 rows x 815 columns]

In [29]:
csv_description

변수명                 변수설명                                내용  \
0           ID               개인 아이디                               NaN   
1          sex                   성별                       1. 남자 2. 여자   
2          age                만나이1)  1~79. 1~79세 80. 80세이상 (2자리수 int)   
3     mh_PHQ_S        (성인) PHQ-9 점수         우울증선별도구 9항목(문항3-1~9) 점수 합   
4       HE_sbp  최종 수축기 혈압(2,3차 평균)                  (3자리수, mmHg, int)   
5       HE_dbp   최종 이완기 혈압(2,3차 평균)                 (3자리수, mmHg, int)   
6        HE_ht                   신장                          □□□.□ cm   
7        HE_wt                   체중                         □□□.□ kg    
8       HE_BMI          체질량지수 (BMI)                       □□□.□ kg/m2   
9   GS_mea_r_1          1차 악력측정_오른손                          □□.□ kg    
10  GS_mea_r_2         2차 악력측정_오른손                           □□.□ kg    
11  GS_mea_r_3         3차 악력측정_오른손                           □□.□ kg    
12  GS_mea_l_1           1차 악력측정_왼손                          □□.□ kg    
13  GS_mea_l_2           2차 악력측정_왼손                          □□.□ kg    
14  GS_mea_l_3           3차 악력측정_왼손                          □□.□ kg    

   필요한가? (Y/N/Not Sure)      type     단위 비고  
0                     Y   Nominal      -  V  
1                     Y   Nominal      -  V  
2                     Y     Ratio      세  V  
3                     Y  Interval    NaN  V  
4                     Y     Ratio   mmHg  V  
5                     Y     Ratio   mmHg  V  
6                     Y     Ratio     cm  V  
7                     Y     Ratio     kg  V  
8                     Y     Ratio  kg/m2  V  
9                     Y     Ratio     kg  V  
10                    Y     Ratio     kg  V  
11                    Y     Ratio     kg  V  
12                    Y     Ratio     kg  V  
13                    Y     Ratio     kg  V  
14                    Y     Ratio     kg  V

In [30]:
csv_all.shape

(7992, 815)

In [31]:
csv_all.columns

Index(['mod_d', 'ID', 'ID_fam', 'year', 'region', 'town_t', 'apt_t', 'psu',
       'sex', 'age',
       ...
       'HEI_TVEG', 'HEI_VEG', 'HEI_PROTF', 'HEI_DAIRY', 'HEI_SFA', 'HEI_NA',
       'HEI_SWEET', 'HEI_CHO', 'HEI_FAT', 'HEI_EN'],
      dtype='object', length=815)

In [32]:
description_columns = csv_description[['변수명']].values.tolist()
key_columns = []
for i in description_columns:
    key_columns.append(i[0])
print(key_columns)

['ID', 'sex', 'age', 'mh_PHQ_S', 'HE_sbp', 'HE_dbp', 'HE_ht', 'HE_wt', 'HE_BMI', 'GS_mea_r_1', 'GS_mea_r_2', 'GS_mea_r_3', 'GS_mea_l_1', 'GS_mea_l_2', 'GS_mea_l_3']


#### 필요한 column만 전체 데이터셋에서 추출

In [33]:
selected_columns = csv_all[key_columns]

In [34]:
selected_columns.head()

ID  sex  age  mh_PHQ_S  HE_sbp  HE_dbp  HE_ht  HE_wt     HE_BMI  \
0  A701020501    1   80       0.0   150.0    84.0  162.4   56.0  21.233226   
1  A701021301    1   78       3.0   132.0    77.0  167.7   76.4  27.166096   
2  A701021302    2   75       2.0   156.0    73.0  157.7   53.2  21.391845   
3  A701022601    2   43       4.0   102.0    70.0  151.0   53.5  23.463883   
4  A701023303    2   80       0.0   149.0    81.0  146.1   57.6  26.984977   

   GS_mea_r_1  GS_mea_r_2  GS_mea_r_3  GS_mea_l_1  GS_mea_l_2  GS_mea_l_3  
0        19.2        22.2        20.0        16.8        18.0        19.6  
1        40.5        38.2        40.0        35.9        37.2        39.2  
2        15.3        17.4        15.3        10.3        10.7        10.4  
3        24.8        24.5        23.8        19.2        19.9        22.6  
4         8.6        13.5        16.1        14.6        13.4        15.0

#### 데이터셋 결측치 확인 및 제거

In [35]:
selected_columns.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7992 entries, 0 to 7991
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   ID          7992 non-null   object 
 1   sex         7992 non-null   int64  
 2   age         7992 non-null   int64  
 3   mh_PHQ_S    5918 non-null   float64
 4   HE_sbp      6847 non-null   float64
 5   HE_dbp      6847 non-null   float64
 6   HE_ht       7593 non-null   float64
 7   HE_wt       7635 non-null   float64
 8   HE_BMI      7591 non-null   float64
 9   GS_mea_r_1  6630 non-null   float64
 10  GS_mea_r_2  6628 non-null   float64
 11  GS_mea_r_3  6626 non-null   float64
 12  GS_mea_l_1  6641 non-null   float64
 13  GS_mea_l_2  6639 non-null   float64
 14  GS_mea_l_3  6637 non-null   float64
dtypes: float64(12), int64(2), object(1)
memory usage: 936.7+ KB


총 7992개의 데이터셋에서 demographics에 해당하는 attribute(ID, sex, age)를 제외하고 전부 결측치 존재.

#### 1. mh_PHQ_S: 본 프로젝트에서 key attribute에 해당하는 값이기 때문에, 결측치가 있는 데이터는 그냥 제거 (Aggregation)

In [36]:
dataset = selected_columns.dropna(subset=['mh_PHQ_S'])
dataset.loc[dataset['mh_PHQ_S'].isna()] # 결측치 데이터 없음 확인

Empty DataFrame
Columns: [ID, sex, age, mh_PHQ_S, HE_sbp, HE_dbp, HE_ht, HE_wt, HE_BMI, GS_mea_r_1, GS_mea_r_2, GS_mea_r_3, GS_mea_l_1, GS_mea_l_2, GS_mea_l_3]
Index: []

mh_PHQ_S 값의 경우, 아래와 같이 카테고리가 구분됨.
각 열별로 0부터 3까지 인덱스를 할당해서, 새로운 depression_scale 열을 생성한 뒤 데이터를 전처리함.
실제 분석에서는 새롭게 생성한 열을 사용할 예정.

| mh_PHQ_S | 1-4점  | 5-9점    | 10-19점  | 20점 이상 |
|----------|-------|---------|---------|--------|
| scale    | 보통 상태 | 가벼운 우울증 | 중간정도 우울증 | 심한 우울증 |
| index    | 0     | 1       | 2       | 3      |

In [37]:
def calculate_depression_scale(row):
    if row['mh_PHQ_S'] < 5:
        return 0
    elif row['mh_PHQ_S'] < 10:
        return 1
    elif row['mh_PHQ_S'] < 20:
        return 2
    else:
        return 3

dataset['depression_scale'] = dataset.apply(calculate_depression_scale, axis=1)
dataset = dataset.drop('mh_PHQ_S', axis=1)

/var/folders/5n/t6zwxly903q0zxv_m_cjvv5c0000gn/T/ipykernel_16945/1418116218.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset['depression_scale'] = dataset.apply(calculate_depression_scale, axis=1)


#### 2. HE_sbp, HE_dbp: 측정 값이 있으면 두 데이터 모두 존재 하는 것으로 보임. 혈압은 다른 데이터로 유추할 수 없기 때문에 그냥 제거

In [38]:
dataset = dataset.dropna(subset=['HE_sbp','HE_dbp'])
dataset.loc[dataset['HE_sbp'].isna()] # 결측치 데이터 없음 확인

Empty DataFrame
Columns: [ID, sex, age, HE_sbp, HE_dbp, HE_ht, HE_wt, HE_BMI, GS_mea_r_1, GS_mea_r_2, GS_mea_r_3, GS_mea_l_1, GS_mea_l_2, GS_mea_l_3, depression_scale]
Index: []

In [39]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5889 entries, 0 to 7988
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ID                5889 non-null   object 
 1   sex               5889 non-null   int64  
 2   age               5889 non-null   int64  
 3   HE_sbp            5889 non-null   float64
 4   HE_dbp            5889 non-null   float64
 5   HE_ht             5849 non-null   float64
 6   HE_wt             5883 non-null   float64
 7   HE_BMI            5847 non-null   float64
 8   GS_mea_r_1        5699 non-null   float64
 9   GS_mea_r_2        5697 non-null   float64
 10  GS_mea_r_3        5695 non-null   float64
 11  GS_mea_l_1        5700 non-null   float64
 12  GS_mea_l_2        5698 non-null   float64
 13  GS_mea_l_3        5696 non-null   float64
 14  depression_scale  5889 non-null   int64  
dtypes: float64(11), int64(3), object(1)
memory usage: 736.1+ KB


#### 3. HE_ht, HE_wt, HE_BMI: HE_ht, HE_wt가 둘다 있으면 HE_BMI를 계산할 수 있음.

HE_ht, HE_wt 둘중 하나라도 결측치가 있으면 제거, 이후에 HE_BMI 결측치의 경우, HE_ht와 HE_wt를 활용해 아래 식으로 계산하여 채우기

BMI = wt/ht^2

* 저체중: BMI < 18.5
* 정상: 18.5 <= BMI < 23
* 비만 전 단계: 23 <= BMI < 25
* 1단계 비만: 25 <= BMI < 30
* 2단계 비만: 30 <= BMI < 35
* 3단계 비만: 35 <= BMI

In [40]:
dataset = dataset.dropna(subset=['HE_ht','HE_wt'])
dataset.loc[dataset['HE_BMI'].isna()] # 결측치 데이터 없음 확인

Empty DataFrame
Columns: [ID, sex, age, HE_sbp, HE_dbp, HE_ht, HE_wt, HE_BMI, GS_mea_r_1, GS_mea_r_2, GS_mea_r_3, GS_mea_l_1, GS_mea_l_2, GS_mea_l_3, depression_scale]
Index: []

In [41]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5847 entries, 0 to 7988
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ID                5847 non-null   object 
 1   sex               5847 non-null   int64  
 2   age               5847 non-null   int64  
 3   HE_sbp            5847 non-null   float64
 4   HE_dbp            5847 non-null   float64
 5   HE_ht             5847 non-null   float64
 6   HE_wt             5847 non-null   float64
 7   HE_BMI            5847 non-null   float64
 8   GS_mea_r_1        5665 non-null   float64
 9   GS_mea_r_2        5663 non-null   float64
 10  GS_mea_r_3        5661 non-null   float64
 11  GS_mea_l_1        5663 non-null   float64
 12  GS_mea_l_2        5661 non-null   float64
 13  GS_mea_l_3        5659 non-null   float64
 14  depression_scale  5847 non-null   int64  
dtypes: float64(11), int64(3), object(1)
memory usage: 730.9+ KB


HE_ht, HE_wt, HE_BMI의 데이터 수가 모두 동일함으로 추가적인 결측치 처리 없음

#### 4. BMI, age, sex를 통해 체지방률 계산 (Feature creation)

성인: 1.2 * BMI + 0.23 * 나이 - 10.8 * 성별(남성=1, 여성=0) - 5.4

아동: 1.5 * BMI - 0.7 * 나이 - 3.6 * 성별(남성=1, 여성=0) + 1.4

해당 데이터에서는 
1. age>=19인 경우 성인
2. 남자=1, 여자=2

In [42]:
def calculate_body_fat(row):
    if row['age'] >= 19 & row['sex'] == 1:
        return 1.2 * row['HE_BMI'] + 0.23 * row['age'] - 10.8 - 5.4
    elif row['age'] >= 19 & row['sex'] == 2:
        return 1.2 * row['HE_BMI'] + 0.23 * row['age'] - 5.4
    elif row['age'] < 19 & row['sex'] == 1:
        return 1.5 * row['bmi'] - 0.7 * row['age'] - 3.6 + 1.4
    else:
        return 1.5 * row['bmi'] - 0.7 * row['age'] + 1.4

dataset['body_fat'] = dataset.apply(calculate_body_fat, axis=1)

In [43]:
dataset

ID  sex  age  HE_sbp  HE_dbp  HE_ht  HE_wt     HE_BMI  \
0     A701020501    1   80   150.0    84.0  162.4   56.0  21.233226   
1     A701021301    1   78   132.0    77.0  167.7   76.4  27.166096   
2     A701021302    2   75   156.0    73.0  157.7   53.2  21.391845   
3     A701022601    2   43   102.0    70.0  151.0   53.5  23.463883   
4     A701023303    2   80   149.0    81.0  146.1   57.6  26.984977   
...          ...  ...  ...     ...     ...    ...    ...        ...   
7981  R708028501    1   41   130.0    82.0  178.7   91.0  28.496550   
7982  R708028502    2   35    93.0    63.0  159.8   55.6  21.773149   
7983  R708028503    2   66    98.0    63.0  156.9   63.9  25.957028   
7987  R708028902    2   73   136.0    82.0  145.2   51.8  24.569512   
7988  R708029201    1   43   122.0    99.0  176.8   84.1  26.904906   

      GS_mea_r_1  GS_mea_r_2  GS_mea_r_3  GS_mea_l_1  GS_mea_l_2  GS_mea_l_3  \
0           19.2        22.2        20.0        16.8        18.0        19.6   
1           40.5        38.2        40.0        35.9        37.2        39.2   
2           15.3        17.4        15.3        10.3        10.7        10.4   
3           24.8        24.5        23.8        19.2        19.9        22.6   
4            8.6        13.5        16.1        14.6        13.4        15.0   
...          ...         ...         ...         ...         ...         ...   
7981        55.5        58.6        60.1        57.4        60.6        58.0   
7982        31.9        28.4        29.3        26.9        24.2        25.1   
7983        18.5        21.0        20.9        18.7        18.4        18.7   
7987        19.7        26.1        23.7        20.9        22.5        20.7   
7988        32.5        30.7        32.4        35.3        32.9        29.1   

      depression_scale   body_fat  
0                    0  27.679871  
1                    0  34.339315  
2                    0  37.520214  
3                    0  32.646660  
4                    0  45.381972  
...                ...        ...  
7981                 0  27.425860  
7982                 0  28.777779  
7983                 0  40.928434  
7987                 0  40.873414  
7988                 1  25.975887  

[5847 rows x 16 columns]

#### 5. GS_mea_r_#, GS_mea_l_#: 3차례에 걸친 악력 측정이므로 평균값을 내서 사용할 것.

1. 만약 3개 attribute 값이 전부 결측치이면 데이터 삭제
2. 3개 중 하나라도 값이 존재한다면 나머지 값을 하나의 값으로 채우기
3. 3개 중 값이 두 개 존재한다면 나머지 값을 두 개의 평균으로 채우기

결측치 확인

In [44]:
dataset = dataset.dropna(subset=['GS_mea_r_1','GS_mea_r_2','GS_mea_r_3'], how='all')
dataset = dataset.dropna(subset=['GS_mea_l_1','GS_mea_l_2','GS_mea_l_3'], how='all')
dataset.loc[dataset['GS_mea_r_1'].isna() | dataset['GS_mea_r_2'].isna() | dataset['GS_mea_r_3'].isna()
 | dataset['GS_mea_l_1'].isna() | dataset['GS_mea_l_2'].isna() | dataset['GS_mea_l_3'].isna()]

ID  sex  age  HE_sbp  HE_dbp  HE_ht  HE_wt     HE_BMI  \
537   A714032302    2   46   114.0    78.0  160.6   56.8  22.022025   
655   A717026602    2   33   112.0    82.0  169.9   57.6  19.954265   
731   A720022102    2   58   136.0    93.0  160.3   60.1  23.388772   
2439  D701024701    1   58   128.0    84.0  163.8   82.0  30.562302   
3703  H703029102    2   46   106.0    69.0  153.2   56.7  24.158253   
3957  H708029503    1   23   101.0    78.0  181.8   66.0  19.968994   
4454  H719028603    2   26   109.0    79.0  167.7   67.4  23.965902   

      GS_mea_r_1  GS_mea_r_2  GS_mea_r_3  GS_mea_l_1  GS_mea_l_2  GS_mea_l_3  \
537         17.0        14.1        10.9        15.6         9.9         NaN   
655         22.1        21.0        22.9        18.7         NaN         NaN   
731         21.1        19.6         NaN        21.0        19.4         NaN   
2439        40.9         NaN         NaN        42.9        42.0        41.8   
3703        17.6        18.7        21.2        12.0         NaN         NaN   
3957        32.1         NaN         NaN        42.0        39.9        42.7   
4454        23.9        25.8         NaN        19.4        20.1        21.1   

      depression_scale   body_fat  
537                  0  31.606430  
655                  0  26.135117  
731                  2  36.006527  
2439                 0  33.814762  
3703                 0  34.169904  
3957                 0  13.052792  
4454                 0  29.339082

GS_mear_r_# 데이터 처리

In [45]:
non_GS = dataset[dataset['GS_mea_r_1'].isna() | dataset['GS_mea_r_2'].isna() | dataset['GS_mea_r_3'].isna() ]

In [46]:
non_GS_index = non_GS.index # 결측치 남아 있는 데이터의 인덱스 값 추출
non_GS_index

Int64Index([731, 2439, 3957, 4454], dtype='int64')

In [47]:
for key in non_GS_index:
    check_list = [dataset.loc[key, dataset.columns==value].values[0] for value in ['GS_mea_r_1','GS_mea_r_2','GS_mea_r_3']]
    if check_list.count('nan') == 1:
        dataset.loc[key, dataset.columns=='GS_mea_r_3'] = (check_list[0] + check_list[1])/2
    else:
        dataset.loc[key, dataset.columns=='GS_mea_r_2'] = check_list[0]
        dataset.loc[key, dataset.columns=='GS_mea_r_3'] = check_list[0]

dataset[dataset['GS_mea_r_1'].isna() | dataset['GS_mea_r_2'].isna() | dataset['GS_mea_r_3'].isna() ] # 결측치 없음 확인

Empty DataFrame
Columns: [ID, sex, age, HE_sbp, HE_dbp, HE_ht, HE_wt, HE_BMI, GS_mea_r_1, GS_mea_r_2, GS_mea_r_3, GS_mea_l_1, GS_mea_l_2, GS_mea_l_3, depression_scale, body_fat]
Index: []

GS_mea_l_# 데이터 처리

In [48]:
non_GS = dataset[dataset['GS_mea_l_1'].isna() | dataset['GS_mea_l_2'].isna() | dataset['GS_mea_l_3'].isna() ]
non_GS_index = non_GS.index
non_GS_index

Int64Index([537, 655, 731, 3703], dtype='int64')

In [49]:
# GS_mear_l_* 데이터처리
for key in non_GS_index:
    check_list = [dataset.loc[key, dataset.columns==value].values[0] for value in ['GS_mea_l_1','GS_mea_l_2','GS_mea_l_3']]
    if check_list.count('nan') == 1:
        dataset.loc[key, dataset.columns=='GS_mea_l_3'] = (check_list[0] + check_list[1])/2
    else:
        dataset.loc[key, dataset.columns=='GS_mea_l_2'] = check_list[0]
        dataset.loc[key, dataset.columns=='GS_mea_l_3'] = check_list[0]

dataset[dataset['GS_mea_l_1'].isna() | dataset['GS_mea_l_2'].isna() | dataset['GS_mea_l_3'].isna() ]  # 결측치 없음 확인

Empty DataFrame
Columns: [ID, sex, age, HE_sbp, HE_dbp, HE_ht, HE_wt, HE_BMI, GS_mea_r_1, GS_mea_r_2, GS_mea_r_3, GS_mea_l_1, GS_mea_l_2, GS_mea_l_3, depression_scale, body_fat]
Index: []

평균을 내서, 새로운 열 추가

#### 6. 결측치 처리를 끝낸 데이터셋 및 저장

In [50]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5591 entries, 0 to 7988
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ID                5591 non-null   object 
 1   sex               5591 non-null   int64  
 2   age               5591 non-null   int64  
 3   HE_sbp            5591 non-null   float64
 4   HE_dbp            5591 non-null   float64
 5   HE_ht             5591 non-null   float64
 6   HE_wt             5591 non-null   float64
 7   HE_BMI            5591 non-null   float64
 8   GS_mea_r_1        5591 non-null   float64
 9   GS_mea_r_2        5591 non-null   float64
 10  GS_mea_r_3        5591 non-null   float64
 11  GS_mea_l_1        5591 non-null   float64
 12  GS_mea_l_2        5591 non-null   float64
 13  GS_mea_l_3        5591 non-null   float64
 14  depression_scale  5591 non-null   int64  
 15  body_fat          5591 non-null   float64
dtypes: float64(12), int64(3), object(1)
memory

In [51]:
dataset.to_csv('data/preprocessing_hn18.csv')